jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/SamProjectWizard.kt

```diff
@@ -61,6 +61,11 @@ data class SdkBasedSdkSettings(
     val sdk: Sdk?
 ) : SdkSettings
 
+sealed class TemplateParameters {
+    data class LocationBasedTemplate(val location: String) : TemplateParameters()
```

Comment:
shouldn't this own which parameters are included? that means we can more easily add new types.

Reply:
I couldn't decide if I wanted all CLI generation in 1 spot using a sealed class, vs abstract method on TemplateParameter that takes a GeneralCommandLine which makes it harder to find what is the concrete CLI generated.... So went with code readability over flexibility. I dont see SAM adding a 3rd way anytime soon.